src/main.cpp

```diff
@@ -3,14 +3,96 @@
 //  Game of Life
 //
 
+#include <SFML/Graphics.hpp>
 #include <cmath>
+#include <chrono>
+#include <thread>
 #include <iostream>
-
+#include "Grid.hpp"
 
 namespace {
 
+const sf::Vector2<uint32_t> screen_dimensions { 764, 508 };
+
 void run(char** argv) {
-    printf("Hello, world!\n");
+    
+    //init clock
+    std::chrono::time_point<std::chrono::system_clock> start{}, end{};
+    std::chrono::duration<double, std::milli> work_time{}, sleep_time{};
+    
+    const double time_step = 1000.0/30.0; //30 FPS
+    bool paused = false;
+    
+    //some SFML variables for drawing a basic window + background
+    auto window = sf::RenderWindow{sf::VideoMode{screen_dimensions.x, screen_dimensions.y}, "Game of Life v1.0"};
+    sf::RectangleShape background{};
+    background.setSize(static_cast<sf::Vector2<float> >(screen_dimensions));
+    background.setPosition(0, 0);
+    background.setFillColor(sf::Color(20, 20, 30));
+    
+    //declare main grid
+    conway::Grid main_grid = conway::Grid(48, 32);
+    main_grid.set_spacing(12.0f);
+    main_grid.update();
+    
+    //start with a basic random init with 40% cell genesis
+    main_grid.random_init();
+    
+    //game loop
+    while (window.isOpen()) {
+        
+        start = std::chrono::system_clock::now(); //start clock
+        std::chrono::duration<double, std::milli> work_time = start - end;
+        
+        if (work_time.count() < time_step)
+        {
+            std::chrono::duration<double, std::milli> delta_ms(time_step - work_time.count());
+            auto delta_ms_duration = std::chrono::duration_cast<std::chrono::milliseconds>(delta_ms);
+            std::this_thread::sleep_for(std::chrono::milliseconds(delta_ms_duration.count()));
```

Comment:
Is this sleep here because of wanting to limit FPS? Use vsync instead, as I said before somewhere. And use `steady_clock`, not `system_clock`. Basic game loop:

```cpp
using Clock = std::chrono::steady_clock;
using Time = std::chrono::duration<float>;
// ...
auto start = Clock::now();
while (window.isOpen()) {
  // compute dt
  auto now = Clock::now();
  auto dt = Time{now - start};
  start = now;
  
  // poll events
  // ...
  
  // tick game
  tick(dt);
  
  // render
  render();
}
```